File: src/Services/Book/Book.Domain/Entities/Author.cs
using Shared;

namespace Book.Domain.Entities;

public class Author : IBaseEntity
{
    public int Id { get; set; }
    public string? OpenLibraryKey { get; set; }
    public string FirstName { get; set; } = null!;
    public string LastName { get; set; } = null!;
    public string FullName { get; set; } = null!;
    public DateOnly? BirthDate { get; set; }
    public DateOnly? DeathDate { get; set; }
    public string? Biography { get; set; }
    public string? ImageUrl { get; set; }
    public List<Book> Books { get; set; } = [];
    public DateTimeOffset DateCreated { get; set; }
    public DateTimeOffset? DateUpdated { get; set; }
    public DateTimeOffset? DateDeleted { get; set; }
}

File: src/Client/Angular.Client/src/app/models/author.ts
export interface Author {
  id: number;
  openLibraryKey?: string | null;
  firstName: string;
  lastName: string;
  fullName: string;
  birthDate?: string | null;
  deathDate?: string | null;
  biography?: string | null;
  imageUrl?: string | null;
}
